src/App.tsx

```diff
@@ -1,24 +1,216 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { FilterType } from './enums/FilterType';
+import { ErrorMessage } from './enums/ErrorMessage';
+
+import { getVisibleTodos } from './utils/getVisibleTodos';
+
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+
+import { ListFilter } from './components/ListFilter';
+import { TodoList } from './components/TodosList';
+import { Header } from './components/Header';
+import { TodoListItem } from './components/TodoListItem';
+import { ErrorModal } from './components/ErrorModal';
+
+import { USER_ID } from './constants';
+
+import { getCompletedTodos } from './utils/getCompletedTodos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.ALL);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessage.NONE);
+  const [isTodoAdding, setIsTodoAdding] = useState(false);
+  const [loadingTodosIds, setloadingTodosIds] = useState<number[]>([]);
```

Comment:
Change the variable name `setloadingTodosIds` to `setLoadingTodosIds` for better readability